adds docs for resample,  MODIS projection (#730)
rafaqz/Rasters.jl: docs/src/resample_proj.md

> ## resample and Projections with ProjString

Reviewer: ```suggestion
## `resample` and projections with `ProjString`

> Geospatial datasets will come in different [projections](https://proj.org/en/9.4/operations/projections/index.html) for whaever reasons. Here, we will focus in some of the most used ones, e.g., `MODIS SINUSOIDAL` and `EPSG`, as well as transformations between them.

Reviewer: ```suggestion
Geospatial datasets will come in different [projections](https://proj.org/en/9.4/operations/projections/index.html) or coordinate reference systems (CRS) for many reasons. Here, we will focus on `MODIS SINUSOIDAL` and `EPSG`, and transformations between them.
```

Let's start by loading the neccesary packages:

````@example modis
using Rasters, RasterDataSources, ArchGDAL
using DimensionalData
using DimensionalData.Lookups
using NaNStatistics
using CairoMakie
````

and let's load a test raster

````@example modis
ras = Raster(WorldClim{BioClim}, 5)
ras_m = replace_missing(ras, missingval=NaN)
````

and let's also take a look

````@example modis
fig = plot(ras_m; colorrange=(0,100))
````

### MODIS SINUSOIDAL PROJECTION

````@example
# ? is this the right ProjString ?, do we need to shift lat, lon?

> SINUSOIDAL_CRS = ProjString("+proj=sinu +lon_0=0 +x_0=0 +y_0=0 +a=6371007.181 +b=6371007.181 +units=m +no_defs")

Reviewer: Users will be wondering what the hell a ProjString is, we should explain that

Reviewer: also @lazarusA why are you defining the ellipse here?  That seems like a step too far.  How about:

```julia
SINUSOIDAL_CRS = ProjString("+proj=sinu +lon_0=0 +type=crs")

Author: Those tiles come/are set with very specific numbers. I think I did some tests, and you not always get the same numbers. BTW, feel free to merge this into your PR, I will probably not finish this anytime soon.

````

and hence the `resample` is performed with

````@example modis
ras_sin = resample(ras_m; size=(1440,720), crs=SINUSOIDAL_CRS, method="sum")
````

let's compare the total counts!

````@example modis
nansum(ras_m), nansum(ras_sin)
````

and, how does this looks like?

````@example modis
fig = plot(ras_sin; colorrange=(0,100))
````

now, let's go back to `latitude` and `longitude`

````@example modis
# ? also here, do we need to shift X, Y?
ras_epsg = resample(ras_sin; size=(1440,720), crs=EPSG(4326), method="sum")
````

> and let's apply `shiftlocus` such that we can harmonize coordinates, which might be needed when building bigger datasets:

Reviewer: what does this do, what does it mean to harmonize coordinates?

Author: I usually need to match to other variables, which are in a specific Lon/lat range.

````@example modis

> locus_resampled = DimensionalData.shiftlocus(Center(), ras_epsg)

Reviewer 2: What happens if we delete these? would be nice to avoid

````

and compare the total counts!

````@example modis
nansum(ras_m), nansum(locus_resampled)
````

````@example modis
fig = plot(ras_epsg; colorrange=(0,100))
````

> ### Assemble a Raster from scratch nativily in the sinusoidal projection

Reviewer: ```suggestion
### Construct a Raster from scratch natively in the sinusoidal projection
